syntax = "proto3";
package afs_operation;


message InitialiseRequest{
    string code_to_initialise = 1;
}
message InitialiseResponse{
    string input_path = 1;
    string output_path = 2;
}

message FileRequest {
    string filename = 1;
    int32 path_select = 2;
    int64 timestamp = 3;
    string content = 4;
}

message FileResponse {
    bytes content = 1;
    int32 length = 2;
    int64 timestamp = 3;
    int32 update_bit =4;  // update = 1 -> needs to update content on the client otherwise no
}

service operators{
    rpc request_dir (InitialiseRequest) returns (InitialiseResponse);
    rpc open (FileRequest) returns (stream FileResponse);
    rpc close (stream FileRequest) returns (FileResponse);
    rpc compare (FileRequest) returns (stream FileResponse);
}